src/ansys/fluent/core/search.py

```diff
@@ -281,10 +280,18 @@ def _write_api_tree_file(api_tree_data: dict, api_object_names: list):
                 if sys.version_info[0] < 3
                 else wn.synsets(name, lang="eng")
             )
-            synset_names = []
+            synset_names = set()
             for api_object_name_synset in api_object_name_synsets:
-                synset_names.append(api_object_name_synset.name().split(".")[0])
-            all_api_object_name_synsets[name] = synset_names
+                synset_names.add(api_object_name_synset.name().split(".")[0])
+            if synset_names:
+                unrelated_synsets = set()
+                synset_names.discard(name)
+                for synset_name in synset_names:
+                    if synset_name not in api_object_names:
+                        unrelated_synsets.add(synset_name)
+                all_api_object_name_synsets[name] = list(
+                    synset_names.difference(unrelated_synsets)
+                )
```

Comment:
No.

Reply:
I mean everything inside the if block only.

Reply:
@hpohekar 
```Python
>>> def intersect_1(synset_names, api_object_names):
...     unrelated_synsets = set()
...     for synset_name in synset_names:
...             if synset_name not in api_object_names:
...                     unrelated_synsets.add(synset_name)
...     return list(
...             synset_names.difference(unrelated_synsets)
...     )
...
...
>>> def intersect_2(synset_names, api_object_names):
...     return list(
...             synset_names.intersection(api_object_names)
...     )
...
...
>>> data = [
... [{"a", "b", "c"}, {"b", "c"}],
... [{"a", "b", "c"}, {"b", "c", "d", "e"}],
... [{"b", "c", "d", "e"}, {"a", "b", "c"}],
... [{"a", "b", "c"}, {"a", "b", "c"}],
...  ]
>>>
>>> for d in data:
...   assert intersect_1(d[0], d[1]) == intersect_2(d[0], d[1])
...
```

Reply:
Thanks for the confirmation.

Reply:
Try to be more flexible in the way you think about review comments. I'm not making random comments here.

Reply:
@seanpearsonuk 

I have tested it as follows but there was a typo in following code snippet. This should contains `x.intersection(api_object_name)` instead of `x.intersection(y)`. Therefore I said it's not equivalent, but I realized it later.

![image](https://github.com/user-attachments/assets/592a44bf-21c7-463b-8dd3-74aadcecbad3)

Reply:
Done. Thank you.

Reply:
as per the latest discussion we don't need this code.